statements.go

```diff
@@ -9,102 +9,49 @@ import (
 
 	"cloud.google.com/go/spanner"
 	"cloud.google.com/go/spanner/admin/database/apiv1/databasepb"
+	"github.com/googleapis/go-sql-spanner/parser"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 )
 
-type parsedStatement interface {
-	parse(parser *statementParser, query string) error
-	executableStatement(c *conn) *executableClientSideStatement
+type executableStatement interface {
+	execContext(ctx context.Context, c *conn, opts *ExecOptions) (driver.Result, error)
+	queryContext(ctx context.Context, c *conn, opts *ExecOptions) (driver.Rows, error)
 }
 
-func parseStatement(parser *statementParser, keyword, query string) (parsedStatement, error) {
```

Comment:
Everything related to parsing in this file has been moved into the `statements.go` file in the `parser` package. This file now only contains the part that is needed to make a parsed statement executable. This separates the actual parsing from how to execute the statement, and allows statements to be parsed without necessarily having an executor at all.